torch/_dynamo/variables/builder.py

```diff
@@ -1498,6 +1500,135 @@ def wrap_symint(self, value):
 
         return unspec_var
 
+    def wrap_symfloat(self, value):
+        # SymFloat wrapping is special.  We first wrap it in the same way we
+        # do an unspecialized primitive, and then we item() it into a
+        # SymFloat.  Removal of the item() call is left to a later FX pass,
+        # mostly because that pass is more easily done after decomposition
+        # (Dynamo doesn't do decomposition right now).
+
+        if self.name in self.tx.output.unspec_variable_map:
+            return self.tx.output.unspec_variable_map[self.name]
+
+        # NB: we specialize on nan input, because our guard modeling in
+        # ShapeEnv cannot deal with nan
+        if (
+            torch._dynamo.config.specialize_float
+            or is_constant_source(self.get_source())
+            or math.isnan(value)
```

Comment:
Can you test that the guard works?

Reply:
What about `0.0` and `1.0`? Perhaps this question goes in line with "what happens when we mix these with a symint?

Reply:
The NaN guard doesn't work (see above.) 0.0/1.0 is not necessary though, I don't see any reason to prophylactically specialize on those values.

Reply:
I was thinking on what happens if you do `int(symfloat)`, but you'll also have issues with any value in the range `(-1, 2)`.

Reply:
What specifically would go wrong with int(symfloat)? Seems fine to me?